pkg/auth/auth.go

```diff
@@ -45,6 +46,41 @@ var (
 	}
 )
 
+// AuthenticatorKey is the hash key used to get a specific authenticator
+type AuthenticatorKey struct {
+	// Issuer is the issuer of the token to match for a speific authenticator.
+	// This value is *required*
+	Issuer string
+	// Audience is the value to match to the token 'aud' which maps to a
+	// specific authenticator.
+	// This value is (optional)
+	Audience string
```

Comment:
Add a comment here that the caller should use `Audience` as the `client-id` or oidc authenticators

Reply:
Good point @pureneelesh. After reading [RFC 7519/aud](https://www.rfc-editor.org/rfc/rfc7519#section-4.1.3), it says that the _token_ can have an `aud` with multiple strings, but the server/principal must identify itself with -a- (meaning one I am assuming) value of the values of the array in aud. But `aud` _can_ be a single string, but generically it should be an array.

What that, I will see about changing the claims to have Audience be a `[]string`, but the API is still the same. Where only one `(issuer, aud value)` tuple applies to a single authenticator. The AuthenticatorManager just needs to check for each of the values of the `aud` to see if any apply.

I will make that change.